File: app/src/main/java/com/example/mvp_practice1/Contract/MainActivityContract.java
package com.example.mvp_practice1.Contract;

import android.content.Context;
import android.content.Intent;
import android.widget.EditText;
import android.widget.TextView;

public interface MainActivityContract {

    interface View{
        void showInfo(String info);
        void showError(String error);
    }

    interface Presenter{
        void saveDetails(
                String firstName,
                String lastName,
                String address,
                String age);
        void loadDetails();
        void goToLogin(Context ctx);
    }
}

File: app/src/main/java/com/example/mvp_practice1/Contract/MainActivityContract.kt
package com.example.mvp_practice1.Contract

import android.content.Context
import android.content.Intent
import android.widget.EditText
import android.widget.TextView

interface MainActivityContract {

    interface View {
        fun showInfo(info: String)
        fun showError(error: String)
    }

    interface Presenter {
        fun saveDetails(
                firstName: String?,
                lastName: String?,
                address: String?,
                age: String?)

        fun loadDetails()
    }
}
